src/free_energy/stats.jl

```diff
@@ -0,0 +1,99 @@
+export 
+    statistical_inefficiency,
+    subsample
+
+@doc """
+    statistical_inefficiency(series::AbstractVector; maxlag::Union{Nothing,Int}=nothing)
+
+Integrated autocorrelation time estimator with IPS truncation and finite-sample taper.
+Returns a NamedTuple: (g, stride, N, N_eff, L).
+- g: statistical inefficiency
```

Comment:
Probably enough fields to make this a struct. There isn't any overhead for that, a named tuple is basically an anonymous struct.